projboard/models/user.py

```diff
@@ -52,9 +52,8 @@ def delete_user_by_nickname(nickname):
         :param nickname: nickname to get the user
         :return:  TRUE/FALSE if the user deleted/not
         """
-        try:
-            user = User.get_user_by_nickname(nickname=nickname)
-            user.delete()
-        except User.DoesNotExist:
+        user = User.get_user_by_nickname(nickname=nickname)
```

Comment:
Please explain the logic here
Is it important to know that user was really deleted or not?
Assume there a request to delete a user which is not in the database.
Is it important to return False? 
the function only delete a user by nickname

Reply:
@Yarboa I'm thinking of changing true or false for test_fail functions, what I assumed that if I generated a fake user which is not in database, and I tried to delete, then the function will return false

Reply:
I think what @Yarboa  is trying to say (and correct me if I am wrong @Yarboa) is that the actual result of a deletion action is that the user is no longer in the DB.  
So, it doesn't really matters if it was existed before or not (you should except the exception and pass it)
The only thing matters is that after the delete action, the user will no longer be in the DB.
So your method may look like:
`
@staticmethod
    def delete_user_by_nickname(nickname):
        user = User.get_user_by_nickname(nickname=nickname)
        try:
             user.delete()
        except Rating.DoesNotExist:
            pass
`

And of course that the tests should be change accordingly

Reply:
@avi-biton About the method you write, the method get_user_by_nickname return None if the nickname has not found in the db. So user.delete() can be problamtic no? 

    `@staticmethod
    def get_user_by_nickname(nickname):

        """
        Method get nickname and return User object if exist, otherwise None
        :param nickname: the nickname to search
        :return: User object
        """
        try:
            user = User.objects.get(nickname=nickname)
        except ObjectDoesNotExist:
            return None
        return user`

Reply:
In addition @Shalevro2 comment, maybe if `get_user_by_nickname(nickname)` will throw User.DoesNotExist exception the function won't make any trouble when using user.delete() because `delete_user_by_nickname(nickname)` would catch the exception